tools/fork-network/src/cli.rs

```diff
@@ -237,30 +237,58 @@ impl ForkNetworkCommand {
         let head = store.get_ser::<Tip>(DBCol::BlockMisc, FINAL_HEAD_KEY)?.unwrap();
         let shard_layout = epoch_manager.get_shard_layout(&head.epoch_id)?;
         let all_shard_uids = shard_layout.get_shard_uids();
-        let fork_head = get_fork_head(&all_shard_uids, store.clone())?;
-        tracing::info!(?fork_head);
+        // Flat state can be at different heights for different shards.
+        // That is fine, we'll simply lookup state root for each .
+        let fork_heads = get_fork_heads(&all_shard_uids, store.clone())?;
+        tracing::info!(?fork_heads);
 
         let chain =
             ChainStore::new(store.clone(), near_config.genesis.config.genesis_height, false);
 
-        let header = chain.get_block_header(&fork_head.hash)?;
-        let epoch_id = header.epoch_id();
-        let num_shards = shard_layout.num_shards();
-        let block_hash = fork_head.hash;
-        let block_height = fork_head.height + 1;
+        // Pick an arbitrary height to include in the new genesis config.
+        let block_height = fork_heads.iter().map(|head| head.height).max().unwrap() + 1;
 
-        let state_roots: Vec<StateRoot> = (0..num_shards)
-            .map(|shard_id| {
-                let shard_uid = epoch_manager.shard_id_to_uid(shard_id, &epoch_id).unwrap();
+        let num_shards = fork_heads.len();
```

Comment:
mini nit: all_shard_uids.len()
I know it's the same but in order to know that I need to understand how get_for_heads works.